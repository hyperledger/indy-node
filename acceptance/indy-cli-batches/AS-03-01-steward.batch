# setup environment
- pool create AS-0301-steward gen_txn_file=./pool_transactions_genesis
pool connect AS-0301-steward
- wallet create AS-03-wallet-steward key=testkey storage_config={"pool":"AS-0301-steward"}
wallet open AS-03-wallet-steward key=testkey
#################
# setup section #
#################
did new seed=TestTrustee100000000000000000001 metadata="test trustee 1"
did new seed=TestSteward100000000000000000001 metadata="test steward 1"
did new seed=TestSteward200000000000000000001 metadata="test steward 2"
did new seed=TestSteward300000000000000000001 metadata="test steward 3"
did new seed=TestTrustAnchor00000000000000001 metadata="test trust anchor 0"
did new seed=TestTrustAnchor10000000000000001 metadata="test trust anchor 1"
did new seed=RandomUser1000000000000000000001 metadata="test identity 1"
did new seed=RandomUser2000000000000000000001 metadata="test identity 2"
# did new seed=RandomUser3000000000000000000001 metadata="test identity 3"
did new seed=RandomUser4000000000000000000001 metadata="test identity 4"
did new seed=RandomUser5000000000000000000001 metadata="test identity 5"
did new seed=000000000000000000000000Trustee1 metadata="default trustee"
did use V4SGRU86Z58d6TV7PBUe6f
# AS Trustee CREATE Steward (steward1)
ledger nym did=7qFmEyYCXcmUFVied5Sp3b role=STEWARD verkey=~Mj3PFUSi6qmrTRonFXHx9n
- ledger get-nym did=7qFmEyYCXcmUFVied5Sp3b
# AS Trustee CREATE Steward (steward3)
ledger nym did=CEJJcfjNGt7YcNLyXaszaq role=STEWARD verkey=~AAGGsLT3yQtTmNdKrHXtC2
- ledger get-nym did=CEJJcfjNGt7YcNLyXaszaq
################
# test section #
################
# #1 AS Steward (steward1) CREATE Trustee (trustee1) (FAIL)
did use 7qFmEyYCXcmUFVied5Sp3b
- ledger nym did=81CCbJwqSyGNLLEtVo1kMq role=TRUSTEE verkey=~UimNgHv7X45jxnh65faK3h
- ledger get-nym did=81CCbJwqSyGNLLEtVo1kMq
# #2.1 AS Steward (steward1) REMOVE Trustee (FAIL)
did use 7qFmEyYCXcmUFVied5Sp3b
- ledger nym did=V4SGRU86Z58d6TV7PBUe6f role=
# #2.2 CHECK Trustee IS VALID
did use V4SGRU86Z58d6TV7PBUe6f
ledger nym did=81CCbJwqSyGNLLEtVo1kMq role=TRUSTEE verkey=~UimNgHv7X45jxnh65faK3h
- ledger get-nym did=81CCbJwqSyGNLLEtVo1kMq
# #3 AS Steward (steward1) CREATE Steward (steward2) (FAIL)
did use 7qFmEyYCXcmUFVied5Sp3b
- ledger nym did=XVP5k4E62PMJJWt2nPacHy role=STEWARD verkey=~Ka6GqJ3qBYRE4Ku9uyFuQW
- ledger get-nym did=XVP5k4E62PMJJWt2nPacHy
# #4.1 AS Steward (steward1) REMOVE Steward (steward3) (FAIL)
did use 7qFmEyYCXcmUFVied5Sp3b
- ledger nym did=CEJJcfjNGt7YcNLyXaszaq role=
# #4.2 CHECK Steward IS VALID
did use CEJJcfjNGt7YcNLyXaszaq
ledger nym did=Bhe7Uh5E1LYLgpLcbuVjj2 role=TRUST_ANCHOR verkey=~NMpYrG7tAXYV4ujYZjddKu
- ledger get-nym did=Bhe7Uh5E1LYLgpLcbuVjj2
# #5.1 AS Steward (steward1) CREATE TrustAnchor (PASS)
did use 7qFmEyYCXcmUFVied5Sp3b
ledger nym did=CDcGtKx1boRYFwPBaGkMmk role=TRUST_ANCHOR verkey=~PuCGfSiTB3NZGi1SH8w7H
- ledger get-nym did=CDcGtKx1boRYFwPBaGkMmk
# #5.2 CHECK TrustAnchor IS VALID
did use CDcGtKx1boRYFwPBaGkMmk
ledger nym did=YUGDShR1RSr5T2CwbM7Hhu verkey=~GfsAf6NpSQDJ1ZWW2X7BiD
- ledger get-nym did=YUGDShR1RSr5T2CwbM7Hhu
# #6.1 AS Steward (steward1) REMOVE TrustAnchor (PASS) ?????????????????????????????????????????????????????????????????
did use 7qFmEyYCXcmUFVied5Sp3b
- ledger nym did=CDcGtKx1boRYFwPBaGkMmk role=
# #6.2 CHECK TrustAnchor IS INVALID
did use CDcGtKx1boRYFwPBaGkMmk
- ledger nym did=Jt7aMnw77aoaBMyhXUNjtt verkey=~DT5pLP1wcvsgAzM78sqiRJ
- ledger get-nym did=Jt7aMnw77aoaBMyhXUNjtt
# #7.1 AS Steward (steward1) RESTORE TrustAnchor (PASS) ?????????????????????????????????????????????????????????????????
did use 7qFmEyYCXcmUFVied5Sp3b
ledger nym did=CDcGtKx1boRYFwPBaGkMmk role=TRUST_ANCHOR
# #7.2 CHECK TrustAnchor IS VALID
did use CDcGtKx1boRYFwPBaGkMmk
ledger nym did=XkZJxs6Uadv6MQeKGGZdZ6 verkey=~Aza4zyTRazcVsokmqNJfsg
- ledger get-nym did=XkZJxs6Uadv6MQeKGGZdZ6
# #8 AS Steward (steward1) CREATE IdentityOwner (PASS)
did use 7qFmEyYCXcmUFVied5Sp3b
ledger nym did=Xm3b3LtJ3UoL5KeYT7ti7j verkey=~GmTyyvE4eHWeSWuiAtmE19
- ledger get-nym did=Xm3b3LtJ3UoL5KeYT7ti7j
# #9.1 AS Steward (steward1) REMOVE self (FAIL)
did use 7qFmEyYCXcmUFVied5Sp3b
- ledger nym did=7qFmEyYCXcmUFVied5Sp3b role=
# #9.2 CHECK Steward IS VALID
ledger nym did=Jt7aMnw77aoaBMyhXUNjtt verkey=~DT5pLP1wcvsgAzM78sqiRJ
- ledger get-nym did=Jt7aMnw77aoaBMyhXUNjtt
### Additional cases ###
did use V4SGRU86Z58d6TV7PBUe6f
did new seed=RandomAttribOwner000000000000001
ledger nym did=SvXt2QGwZF1kXTcpd2pJ37 verkey=~TmkMNbfcATrfJopaq4KcdV role=STEWARD
did new seed=RandomUserForRotateKey0000000001
ledger nym did=6LKnRH6hWPSpoWu824s5JH verkey=~JyFXPqScFNBSoPG4cgTeb1 role=STEWARD
# Only owner can edit existing NYMs
did use SvXt2QGwZF1kXTcpd2pJ37
did rotate-key seed=RandomUserForRotateKey0NewKey001
# Not owner can't edit existing NYMs (FAIL)
did use 6LKnRH6hWPSpoWu824s5JH
- ledger nym did=SvXt2QGwZF1kXTcpd2pJ37 verkey=~TmkMNbfcATrfJopaq4KcdV
# Only owners can create ATTRIBs
did use SvXt2QGwZF1kXTcpd2pJ37
ledger attrib did=SvXt2QGwZF1kXTcpd2pJ37 raw={"endpoint":{"ha":"127.0.0.1:5555"}}
# Only owners can edit ATTRIBs
ledger attrib did=SvXt2QGwZF1kXTcpd2pJ37 raw={"endpoint":{"ha":"127.0.0.1:6666"}}
# Not owner can't create attrib (FAIL)
did use SvXt2QGwZF1kXTcpd2pJ37
- ledger attrib did=6LKnRH6hWPSpoWu824s5JH raw={"endpoint":{"ha":"127.0.0.1:5555"}}
# Not owner can't edit attrib (FAIL)
did use 6LKnRH6hWPSpoWu824s5JH
- ledger attrib did=SvXt2QGwZF1kXTcpd2pJ37 raw={"endpoint":{"ha":"127.0.0.1:5555"}}
# Steward can create SCHEMA:
ledger schema name=StewardSchema version=1.0 attr_names=name,age
# Steward can create CLAIM_DEF:
ledger cred-def schema_id=1 signature_type=CL primary={"n":"1","s":"2","rms":"3","r":{"age":"4","name":"5"},"rctxt":"6","z":"7"} tag=1