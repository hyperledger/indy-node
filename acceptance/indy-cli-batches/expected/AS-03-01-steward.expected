- pool create AS-0301-steward gen_txn_file=./pool_transactions_genesis
[1;32mPool config "AS-0301-steward" has been created[0m

pool connect AS-0301-steward
[1;32mPool "AS-0301-steward" has been connected[0m

- wallet create AS-03-wallet-steward key=testkey storage_config={"pool":"AS-0301-steward"}
[1;32mWallet "AS-03-wallet-steward" has been created[0m

wallet open AS-03-wallet-steward key=testkey
[1;32mWallet "AS-03-wallet-steward" has been opened[0m

did new seed=TestTrustee100000000000000000001 metadata="test trustee 1"
[1;32mDid "81CCbJwqSyGNLLEtVo1kMq" has been created with "~UimNgHv7X45jxnh65faK3h" verkey[0m
[1;32mMetadata has been saved for DID "81CCbJwqSyGNLLEtVo1kMq"[0m

did new seed=TestSteward100000000000000000001 metadata="test steward 1"
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been created with "~Mj3PFUSi6qmrTRonFXHx9n" verkey[0m
[1;32mMetadata has been saved for DID "7qFmEyYCXcmUFVied5Sp3b"[0m

did new seed=TestSteward200000000000000000001 metadata="test steward 2"
[1;32mDid "XVP5k4E62PMJJWt2nPacHy" has been created with "~Ka6GqJ3qBYRE4Ku9uyFuQW" verkey[0m
[1;32mMetadata has been saved for DID "XVP5k4E62PMJJWt2nPacHy"[0m

did new seed=TestSteward300000000000000000001 metadata="test steward 3"
[1;32mDid "CEJJcfjNGt7YcNLyXaszaq" has been created with "~AAGGsLT3yQtTmNdKrHXtC2" verkey[0m
[1;32mMetadata has been saved for DID "CEJJcfjNGt7YcNLyXaszaq"[0m

did new seed=TestTrustAnchor00000000000000001 metadata="test trust anchor 0"
[1;32mDid "CDcGtKx1boRYFwPBaGkMmk" has been created with "~PuCGfSiTB3NZGi1SH8w7H" verkey[0m
[1;32mMetadata has been saved for DID "CDcGtKx1boRYFwPBaGkMmk"[0m

did new seed=TestTrustAnchor10000000000000001 metadata="test trust anchor 1"
[1;32mDid "Bhe7Uh5E1LYLgpLcbuVjj2" has been created with "~NMpYrG7tAXYV4ujYZjddKu" verkey[0m
[1;32mMetadata has been saved for DID "Bhe7Uh5E1LYLgpLcbuVjj2"[0m

did new seed=RandomUser1000000000000000000001 metadata="test identity 1"
[1;32mDid "XkZJxs6Uadv6MQeKGGZdZ6" has been created with "~Aza4zyTRazcVsokmqNJfsg" verkey[0m
[1;32mMetadata has been saved for DID "XkZJxs6Uadv6MQeKGGZdZ6"[0m

did new seed=RandomUser2000000000000000000001 metadata="test identity 2"
[1;32mDid "Xm3b3LtJ3UoL5KeYT7ti7j" has been created with "~GmTyyvE4eHWeSWuiAtmE19" verkey[0m
[1;32mMetadata has been saved for DID "Xm3b3LtJ3UoL5KeYT7ti7j"[0m

did new seed=RandomUser4000000000000000000001 metadata="test identity 4"
[1;32mDid "YUGDShR1RSr5T2CwbM7Hhu" has been created with "~GfsAf6NpSQDJ1ZWW2X7BiD" verkey[0m
[1;32mMetadata has been saved for DID "YUGDShR1RSr5T2CwbM7Hhu"[0m

did new seed=RandomUser5000000000000000000001 metadata="test identity 5"
[1;32mDid "Jt7aMnw77aoaBMyhXUNjtt" has been created with "~DT5pLP1wcvsgAzM78sqiRJ" verkey[0m
[1;32mMetadata has been saved for DID "Jt7aMnw77aoaBMyhXUNjtt"[0m

did new seed=000000000000000000000000Trustee1 metadata="default trustee"
[1;32mDid "V4SGRU86Z58d6TV7PBUe6f" has been created with "~CoRER63DVYnWZtK8uAzNbx" verkey[0m
[1;32mMetadata has been saved for DID "V4SGRU86Z58d6TV7PBUe6f"[0m

did use V4SGRU86Z58d6TV7PBUe6f
[1;32mDid "V4SGRU86Z58d6TV7PBUe6f" has been set as active[0m

ledger nym did=7qFmEyYCXcmUFVied5Sp3b role=STEWARD verkey=~Mj3PFUSi6qmrTRonFXHx9n
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| V4SGRU86Z58d6TV7PBUe6f | 33              | 1534005195103816786 | 2018-08-11 16:33:15 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+-------------------------+---------+
| Did                    | Verkey                  | Role    |
+------------------------+-------------------------+---------+
| 7qFmEyYCXcmUFVied5Sp3b | ~Mj3PFUSi6qmrTRonFXHx9n | STEWARD |
+------------------------+-------------------------+---------+

- ledger get-nym did=7qFmEyYCXcmUFVied5Sp3b
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| V4SGRU86Z58d6TV7PBUe6f | 33              | 1534005195894891570 | 2018-08-11 16:33:15 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+-------------------------+---------+
| Identifier             | Dest                   | Verkey                  | Role    |
+------------------------+------------------------+-------------------------+---------+
| V4SGRU86Z58d6TV7PBUe6f | 7qFmEyYCXcmUFVied5Sp3b | ~Mj3PFUSi6qmrTRonFXHx9n | STEWARD |
+------------------------+------------------------+-------------------------+---------+

ledger nym did=CEJJcfjNGt7YcNLyXaszaq role=STEWARD verkey=~AAGGsLT3yQtTmNdKrHXtC2
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| V4SGRU86Z58d6TV7PBUe6f | 34              | 1534005195933339924 | 2018-08-11 16:33:16 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+-------------------------+---------+
| Did                    | Verkey                  | Role    |
+------------------------+-------------------------+---------+
| CEJJcfjNGt7YcNLyXaszaq | ~AAGGsLT3yQtTmNdKrHXtC2 | STEWARD |
+------------------------+-------------------------+---------+

- ledger get-nym did=CEJJcfjNGt7YcNLyXaszaq
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| V4SGRU86Z58d6TV7PBUe6f | 34              | 1534005196874579207 | 2018-08-11 16:33:16 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+-------------------------+---------+
| Identifier             | Dest                   | Verkey                  | Role    |
+------------------------+------------------------+-------------------------+---------+
| V4SGRU86Z58d6TV7PBUe6f | CEJJcfjNGt7YcNLyXaszaq | ~AAGGsLT3yQtTmNdKrHXtC2 | STEWARD |
+------------------------+------------------------+-------------------------+---------+

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

- ledger nym did=81CCbJwqSyGNLLEtVo1kMq role=TRUSTEE verkey=~UimNgHv7X45jxnh65faK3h
[1;31mTransaction has been rejected: STEWARD cannot add TRUSTEE[0m

- ledger get-nym did=81CCbJwqSyGNLLEtVo1kMq
[1;31mNYM not found[0m

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

- ledger nym did=V4SGRU86Z58d6TV7PBUe6f role=
[1;31mTransaction has been rejected: 7qFmEyYCXcmUFVied5Sp3b is neither Trustee nor owner of V4SGRU86Z58d6TV7PBUe6f[0m

did use V4SGRU86Z58d6TV7PBUe6f
[1;32mDid "V4SGRU86Z58d6TV7PBUe6f" has been set as active[0m

ledger nym did=81CCbJwqSyGNLLEtVo1kMq role=TRUSTEE verkey=~UimNgHv7X45jxnh65faK3h
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| V4SGRU86Z58d6TV7PBUe6f | 35              | 1534005198493653092 | 2018-08-11 16:33:19 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+-------------------------+---------+
| Did                    | Verkey                  | Role    |
+------------------------+-------------------------+---------+
| 81CCbJwqSyGNLLEtVo1kMq | ~UimNgHv7X45jxnh65faK3h | TRUSTEE |
+------------------------+-------------------------+---------+

- ledger get-nym did=81CCbJwqSyGNLLEtVo1kMq
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| V4SGRU86Z58d6TV7PBUe6f | 35              | 1534005199941485532 | 2018-08-11 16:33:19 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+-------------------------+---------+
| Identifier             | Dest                   | Verkey                  | Role    |
+------------------------+------------------------+-------------------------+---------+
| V4SGRU86Z58d6TV7PBUe6f | 81CCbJwqSyGNLLEtVo1kMq | ~UimNgHv7X45jxnh65faK3h | TRUSTEE |
+------------------------+------------------------+-------------------------+---------+

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

- ledger nym did=XVP5k4E62PMJJWt2nPacHy role=STEWARD verkey=~Ka6GqJ3qBYRE4Ku9uyFuQW
[1;31mTransaction has been rejected: STEWARD cannot add STEWARD[0m

- ledger get-nym did=XVP5k4E62PMJJWt2nPacHy
[1;31mNYM not found[0m

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

- ledger nym did=CEJJcfjNGt7YcNLyXaszaq role=
[1;31mTransaction has been rejected: 7qFmEyYCXcmUFVied5Sp3b is neither Trustee nor owner of CEJJcfjNGt7YcNLyXaszaq[0m

did use CEJJcfjNGt7YcNLyXaszaq
[1;32mDid "CEJJcfjNGt7YcNLyXaszaq" has been set as active[0m

ledger nym did=Bhe7Uh5E1LYLgpLcbuVjj2 role=TRUST_ANCHOR verkey=~NMpYrG7tAXYV4ujYZjddKu
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| CEJJcfjNGt7YcNLyXaszaq | 36              | 1534005201589633068 | 2018-08-11 16:33:22 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+-------------------------+--------------+
| Did                    | Verkey                  | Role         |
+------------------------+-------------------------+--------------+
| Bhe7Uh5E1LYLgpLcbuVjj2 | ~NMpYrG7tAXYV4ujYZjddKu | TRUST_ANCHOR |
+------------------------+-------------------------+--------------+

- ledger get-nym did=Bhe7Uh5E1LYLgpLcbuVjj2
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| CEJJcfjNGt7YcNLyXaszaq | 36              | 1534005202961457481 | 2018-08-11 16:33:22 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+-------------------------+--------------+
| Identifier             | Dest                   | Verkey                  | Role         |
+------------------------+------------------------+-------------------------+--------------+
| CEJJcfjNGt7YcNLyXaszaq | Bhe7Uh5E1LYLgpLcbuVjj2 | ~NMpYrG7tAXYV4ujYZjddKu | TRUST_ANCHOR |
+------------------------+------------------------+-------------------------+--------------+

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

ledger nym did=CDcGtKx1boRYFwPBaGkMmk role=TRUST_ANCHOR verkey=~PuCGfSiTB3NZGi1SH8w7H
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| 7qFmEyYCXcmUFVied5Sp3b | 37              | 1534005203036368824 | 2018-08-11 16:33:23 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+--------------+
| Did                    | Verkey                 | Role         |
+------------------------+------------------------+--------------+
| CDcGtKx1boRYFwPBaGkMmk | ~PuCGfSiTB3NZGi1SH8w7H | TRUST_ANCHOR |
+------------------------+------------------------+--------------+

- ledger get-nym did=CDcGtKx1boRYFwPBaGkMmk
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| 7qFmEyYCXcmUFVied5Sp3b | 37              | 1534005204011390885 | 2018-08-11 16:33:23 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+------------------------+--------------+
| Identifier             | Dest                   | Verkey                 | Role         |
+------------------------+------------------------+------------------------+--------------+
| 7qFmEyYCXcmUFVied5Sp3b | CDcGtKx1boRYFwPBaGkMmk | ~PuCGfSiTB3NZGi1SH8w7H | TRUST_ANCHOR |
+------------------------+------------------------+------------------------+--------------+

did use CDcGtKx1boRYFwPBaGkMmk
[1;32mDid "CDcGtKx1boRYFwPBaGkMmk" has been set as active[0m

ledger nym did=YUGDShR1RSr5T2CwbM7Hhu verkey=~GfsAf6NpSQDJ1ZWW2X7BiD
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| CDcGtKx1boRYFwPBaGkMmk | 38              | 1534005204050115133 | 2018-08-11 16:33:24 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+-------------------------+------+
| Did                    | Verkey                  | Role |
+------------------------+-------------------------+------+
| YUGDShR1RSr5T2CwbM7Hhu | ~GfsAf6NpSQDJ1ZWW2X7BiD | -    |
+------------------------+-------------------------+------+

- ledger get-nym did=YUGDShR1RSr5T2CwbM7Hhu
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| CDcGtKx1boRYFwPBaGkMmk | 38              | 1534005205042158052 | 2018-08-11 16:33:24 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+-------------------------+------+
| Identifier             | Dest                   | Verkey                  | Role |
+------------------------+------------------------+-------------------------+------+
| CDcGtKx1boRYFwPBaGkMmk | YUGDShR1RSr5T2CwbM7Hhu | ~GfsAf6NpSQDJ1ZWW2X7BiD | -    |
+------------------------+------------------------+-------------------------+------+

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

- ledger nym did=CDcGtKx1boRYFwPBaGkMmk role=
[1;31mTransaction has been rejected: 7qFmEyYCXcmUFVied5Sp3b is neither Trustee nor owner of CDcGtKx1boRYFwPBaGkMmk[0m

did use CDcGtKx1boRYFwPBaGkMmk
[1;32mDid "CDcGtKx1boRYFwPBaGkMmk" has been set as active[0m

- ledger nym did=Jt7aMnw77aoaBMyhXUNjtt verkey=~DT5pLP1wcvsgAzM78sqiRJ
[1;32mNym request has been sent to Ledger.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| From                   | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| CDcGtKx1boRYFwPBaGkMmk | 39              | 1534005205716058459 | 2018-08-11 16:33:26 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+-------------------------+------+
| Did                    | Verkey                  | Role |
+------------------------+-------------------------+------+
| Jt7aMnw77aoaBMyhXUNjtt | ~DT5pLP1wcvsgAzM78sqiRJ | -    |
+------------------------+-------------------------+------+

- ledger get-nym did=Jt7aMnw77aoaBMyhXUNjtt
[1;32mFollowing NYM has been received.[0m
[1;32mMetadata:[0m
+------------------------+-----------------+---------------------+---------------------+
| Identifier             | Sequence Number | Request ID          | Transaction time    |
+------------------------+-----------------+---------------------+---------------------+
| CDcGtKx1boRYFwPBaGkMmk | 39              | 1534005207074672779 | 2018-08-11 16:33:26 |
+------------------------+-----------------+---------------------+---------------------+
[1;32mData:[0m
+------------------------+------------------------+-------------------------+------+
| Identifier             | Dest                   | Verkey                  | Role |
+------------------------+------------------------+-------------------------+------+
| CDcGtKx1boRYFwPBaGkMmk | Jt7aMnw77aoaBMyhXUNjtt | ~DT5pLP1wcvsgAzM78sqiRJ | -    |
+------------------------+------------------------+-------------------------+------+

did use 7qFmEyYCXcmUFVied5Sp3b
[1;32mDid "7qFmEyYCXcmUFVied5Sp3b" has been set as active[0m

ledger nym did=CDcGtKx1boRYFwPBaGkMmk role=TRUST_ANCHOR
[1;31mTransaction has been rejected: 7qFmEyYCXcmUFVied5Sp3b is neither Trustee nor owner of CDcGtKx1boRYFwPBaGkMmk[0m
[1;31mBatch execution failed at line #49[0m
[1;32mWallet "AS-03-wallet-steward" has been closed[0m
[1;32mPool "AS-0301-steward" has been disconnected[0m
[1;32mGoodbye...[0m
